packages/gatsby-plugin-fastify/src/plugins/headers.ts

```diff
@@ -0,0 +1,46 @@
+import fp from "fastify-plugin";
+
+import { SECURITY_HEADERS } from "../utils/constants";
+import { removeQueryParmsFromUrl } from "../utils/routes";
+
+import type { Headers } from "../utils/headers";
+import type { FastifyPluginAsync, FastifyReply } from "fastify";
+
+export const handleHeaders: FastifyPluginAsync<{
+  headers: Headers;
+  useDefaultSecurity: boolean;
+}> = fp(async (fastify, { headers, useDefaultSecurity }) => {
+  fastify.addHook("onSend", async function (request, reply: FastifyReply) {
+    let pageHeaders = {};
```

Comment:
Instead of mutating this, I think [`Object.assign` ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) can clean this up.